### PR TITLE
Split the invoice related emails into its own rule - rename DAOResourceLoader

### PR DESCRIPTION
**Pull request by minsunny**
https://nanopay.atlassian.net/browse/NP-5780
nanopay: https://github.com/nanoPayinc/NANOPAY/pull/13804

Making sendingEmail functionalities in InvoiceNotificationDAO.js into its own rule
That rule is applied PermissionedInvoiceRule which checking payee / payer's permissions depending on the given capablities.
Rename DAOResourceLoader.java to EmailTemplateSupport - Confirmed with @kgrgreer 
Loggers and minor updates in email logic

### CODE REVIEW

File: src/foam/nanos/notification/email/DAOResourceLoader.java [hunk 1]
```diff
@@ -53,7 +55,9 @@ public static EmailTemplate findTemplate(X x, String name, String groupId, Strin
       Y     *     *     *
     */
 
+    logger.info("name : " + name + ", groupId : " + groupId + ", locale : " + locale + ", spid : " + spid);
```
Comment (jlhughes) [inline src/foam/nanos/notification/email/DAOResourceLoader.java]: You don't need to concatenate in log messages. slows things down as well.   Logging is lazy so the string will only be concatenated if actually outputting this log level. 
```suggestion
    logger.info("name", name, "groupId", groupId, "locale", locale, "spid", spid);
```

File: src/foam/nanos/notification/email/DAOResourceLoader.java [hunk 2]
```diff
@@ -16,6 +16,7 @@
 import foam.nanos.auth.Subject;
 import foam.nanos.auth.User;
 import foam.nanos.notification.email.EmailTemplate;
+import foam.nanos.logger.Logger;
```
Comment (jlhughes) [inline src/foam/nanos/notification/email/DAOResourceLoader.java:16]:
```suggestion
import foam.nanos.logger.Logger;
import foam.nanos.logger.Loggers;
```

File: src/foam/nanos/notification/email/DAOResourceLoader.java [hunk 3]
```diff
@@ -42,6 +43,7 @@ public static EmailTemplate findTemplate(X x, String name, String groupId, Strin
     DAO groupDAO = (DAO) x.get("groupDAO");
     DAO emailTemplateDAO = (DAO) x.get("localEmailTemplateDAO");
     EmailTemplate emailTemplate = null;
+    Logger logger = (Logger) x.get("logger");
```
Comment (jlhughes) [inline src/foam/nanos/notification/email/DAOResourceLoader.java]:
```suggestion
    Logger logger = Loggers.logger(x, this);
```
This will create a logger that will prefix log messages this class name so we know where they originated.

File: src/foam/nanos/notification/email/DAOResourceLoader.java [hunk 4]
```diff
@@ -125,21 +132,39 @@ public static EmailTemplate findTemplateHelper(X x, String name, String groupId,
...
     return emailTemplate_;
   }
 
+  public static void setTemplateSource(X x, String name, EmailTemplate emailTemplate, Map templateArgs) {
+      String sourceType = (String) templateArgs.get("templateSourceType");
+      String templateSource = (String) templateArgs.get("templateSource");
+      String source = templateSource != null ? templateSource : "emailTemplate";
+      emailTemplate.setSourceClass(source);
+
+      PM pm = PM.create(x, source,  "emailTemplate: " + name);
```
Comment (jlhughes) [inline src/foam/nanos/notification/email/DAOResourceLoader.java]:
```suggestion
      PM pm = PM.create(x, source,  "emailTemplate", name);
```
PM will concatenate arguments with : for you.

File: src/foam/nanos/notification/email/DAOResourceLoader.java [hunk 5]
```diff
@@ -95,6 +101,9 @@ public static EmailTemplate findTemplate(X x, String name, String groupId, Strin
     if ( emailTemplate == null )
       emailTemplate = findTemplateHelper(x, name, "", "en", "", templateArgs);
 
+    if ( emailTemplate == null )
```
Comment (kgrgreer) [inline src/foam/nanos/notification/email/DAOResourceLoader.java]: Why is this necessary? Doesn't findTemplateHelper() do this same check as its last step?

File: src/foam/nanos/notification/email/DAOResourceLoader.java [hunk 6]
```diff
@@ -16,6 +16,8 @@
 import foam.nanos.auth.Subject;
```
Comment (kgrgreer) [inline src/foam/nanos/notification/email/DAOResourceLoader.java:14]: Can we rename this model from DAOResourceLoader to something more meaningful? This name is from when we used an old 3rd party template library that required us to implement their "ResourceLoader" interface, but that requirement is long-since gone so we could now rename it.

**Comment by kgrgreer** [comment]
LGTM
